### DISCUSSION TITLE
concatenate fields CRUD

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#564] [Q&A (Help)] [answered]
I need to understand on thing..
I've two models Request & User
In Request model i've this:
```php
    public function user()
    {
    	return $this->belongsTo('App\Models\User');
    }
```

In RequestCrudController if use `CRUD::column('user')` really i see password field of user table, so first question is:
How CRUD decide what user field to show ?

Second question:

In user table i've fields first_name and second_name and in my RequestCrudController i'd like to use something like this:
`CRUD::column('FullName')`

I've tried to create an accessor like this:
```php
    public function getFullNameAttribute()
    {
    	return $this->first_name . ' ' . $this->last_name;
    }
```
But i don't know how to use it in CRUD !

Searching on web i've found this old (2017) issue: https://github.com/Laravel-Backpack/CRUD/issues/402

The solutions there are still valid or now that we have Laravel 10.x and BackPack 5.x or 6.x there are better solutions ?

Thanks for help

**Comment by tabacitu** [comment, marked as answer]
Hi @blondie63 ,

> In RequestCrudController if use CRUD::column('user') really i see password field of user table, so first question is:
> How CRUD decide what user field to show ?

Backpack tries to 
- guess the "_identifiable attribute_"; That is, the attribute on your Model that is most likely to be unique and memorable so that users can identify an certain entry from it. Like name, email etc. 
- then based on that attribute, it will try to guess the field type

You can see exactly how `identifiableAttribute` works here: https://github.com/Laravel-Backpack/CRUD/blob/main/src/app/Models/Traits/HasIdentifiableAttribute.php

If it's not correct for your model you can do something like this on your User model and then Backpack won't try to guess it any more:
```php
public $identifiableAttribute = "name";
```

> In user table i've fields first_name and second_name and in my RequestCrudController i'd like to use something like this:
CRUD::column('FullName')
> But i don't know how to use it in CRUD !

You add CRUD columns for accessors, no problem, just do `CRUD::column('full_name')` (snake case, that's how Laravel converts them) and make sure in your User model you have instructed Laravel to always get that attribute when the model is gotten:

```php
public $appends = ['full_name'];
```

Hope it helps.
Cheers!

**Reply by blondie63** [reply]
ok i understood all

Thanks and regards

**Comment by blondie63** [comment]
Hi @tabacitu ,  i need a little more help.. :)
take a look on my schema:
<img width="563" alt="RequestFlowCrud" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/18d89f43-bba9-41dc-aed8-a3924c971eaf">
On RequestFlowCrudController.php i need to have a column RequestType->name (by lang)
The path is request_flows->request_id->requests_user->request_type_id->request_types->name

Is it possible to create something on Models to have this double jump ? :)
Thanks for help

**Reply by tabacitu** [reply]
Yes, if have correct Eloquent relationships on your models, most likely you can specify the name of your column using dot notation, and Backpack will follow the relationship flow. In your case it would probably be `CRUD::column('user.type.name')`

**Comment by blondie63** [comment]
@tabacitu  relations you can see on my screenshot was created by DevTools
you write CRUD::column('user.type.name') but what you with:
user
type
name

fields name ?
functions name ?

for example on my RequestFlow model i've this:
    public function requestUser(): BelongsTo
    {
        return $this->belongsTo(RequestUser::class);
    }
    
and

    public function statusStart(): BelongsTo
    {
        return $this->belongsTo(RequestStatus::class, 'status_id_start');
    }

CRUD::column('statusStart') works fine, using function name as column name

**Reply by blondie63** [reply]
I fixed using this accessor:
    public function getRequestTypeNameAttribute()
    {
        return $this->requestUser->requestType->name;
    }
